orion/pkg/controllers/user_controller_test.go

```diff
@@ -282,16 +282,16 @@ func createMockUser(id uint, firstName, lastName, middleName, email, phone strin
 		Id:         id,
 		FirstName:  firstName,
 		LastName:   lastName,
-		MiddleName: middleName,
+		MiddleName: domains.CreateNullString(middleName),
 		Email:      email,
 		Phone:      phone,
 		IsGuardian: isGuardian,
-		GuardianId: guardianId,
+		GuardianId: domains.CreateNullUint(guardianId),
 	}
 }
 
 func createBodyFromUser(user domains.User) io.Reader {
-	marshal, err := json.Marshal(user)
+	marshal, err := json.Marshal(&user)
```

Comment:
Overall, I think everything looks decently easy to use. It's just this part that kind of scares me because it's so nuanced. When marshaling json, especially a struct with nested structs inside it that has custom Marshaling, you must give it a pointer. `json.Marshal(...)` takes an `interface{}` typed parameter and for some reason (I found a source before, but can't find it atm) it won't call the custom marshaller unless you pass the pointer of the struct instead.
One thing we could do is to make sure we ALWAYS pass in a pointer of a struct to `json.Marshal` so we never bump into this nuanced problem. But yeah - it's nuanced and errors that originate from not including `&` can be misleading at times. :(